core/src/main/java/com/netflix/conductor/service/AdminService.java

```diff
@@ -0,0 +1,97 @@
+package com.netflix.conductor.service;
+
+import com.netflix.conductor.annotations.Trace;
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.core.execution.WorkflowExecutor;
+import com.netflix.conductor.dao.QueueDAO;
+import com.netflix.conductor.service.utils.ServiceUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * @author fjhaveri
+ *
+ */
+@Singleton
+@Trace
+public class AdminService {
+
+    private static Logger logger = LoggerFactory.getLogger(AdminService.class);
+
+    private final Configuration config;
+
+    private final ExecutionService executionService;
+
+    private final QueueDAO queueDAO;
+
+    private String version;
+
+    private String buildDate;
+
+    @Inject
+    public AdminService(Configuration config, ExecutionService executionService, QueueDAO queueDAO) {
+        this.config = config;
+        this.executionService = executionService;
+        this.queueDAO = queueDAO;
+        this.version = "UNKNOWN";
+        this.buildDate = "UNKNOWN";
+
+        try {
+            InputStream propertiesIs = this.getClass().getClassLoader().getResourceAsStream("META-INF/conductor-core.properties");
+            Properties prop = new Properties();
+            prop.load(propertiesIs);
+            this.version = prop.getProperty("Implementation-Version");
+            this.buildDate = prop.getProperty("Build-Date");
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
```

Comment:
Please log better error message here:
```java
logger.error("Error loading properties", e);
```

Reply:
I copy pasted the existing code. I will avoid modifying this in this PR.